### DISCUSSION TITLE
Multiple icons

### DISCUSSION THREAD

**Discussion by Matesax** [HavenDV/H.NotifyIcon#59] [Q&A] [answered]
Hi,
starting simple WPF appp is generating multiple systrays and after close some are unremovable. I have keep to restart exporer.exe ... I've tried to set Id, but it somehow crushed on taskbar icon. Without ID application is working, but it filling systray with dead icons ... (Really only restart of exporer.exe is working.) With 2 or 3 icons generated on sartup only one is working with current application. Otherwise everything working well.

`<Window x:Class="TestService.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:tb="clr-namespace:H.NotifyIcon;assembly=H.NotifyIcon.Wpf"
        xmlns:local="clr-namespace:TestService"
        mc:Ignorable="d"
        Title="TestService" Height="450" Width="800"
        Icon="/TestService;component/EximManager.ico"
        StateChanged="Window_StateChanged">
    <StackPanel>
        <tb:TaskbarIcon TrayMouseDoubleClick="Notify_TrayMouseDoubleClick" ToolTipText="TestService" Name="Notify" IconSource="/TestService;component/EximManager.ico">
            <tb:TaskbarIcon.ContextMenu>
                <ContextMenu>
                    <MenuItem Header="Otevřít" Click="Notify_TrayMouseDoubleClick" />
                    <MenuItem Header="Ukončit" Click="MenuItemClose_Click" />
                </ContextMenu>
            </tb:TaskbarIcon.ContextMenu>
        </tb:TaskbarIcon>
        <Grid>
            <Grid.ColumnDefinitions>
....
</Window>`

Fatal crash if ID is set:

> You need to find TaskbarIcon.cs to view the source for the current call stack frame

**Comment by Matesax** [comment]
Nice. With ID set I've got 2 unremovable icons and restart of explorer.exe is not working for them ...

**Comment by HavenDV** [comment, marked as answer]
Hi. 
First of all, I want to note that we should call Dispose () before exiting the application in order to correctly clear the tray icons.
Second, use CustomName instead of Id for the second and subsequent icons.

**Reply by Matesax** [reply]
Hi,
as I wrote, unwanted multiplication is occuring since startup. So yeap - Dispose is great suggestion, Understandable. I will definitely try CustomName. But I really want only one icon and multiplication was always unwanted. Missing dispose maby keepem dead icon in backgroudn and next startup is making the multiplication - maby ...

**Reply by Matesax** [reply]
Same error for CustomName as for ID. But Dispose seems to solve startup multiplication.
